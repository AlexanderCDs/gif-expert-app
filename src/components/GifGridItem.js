/**
 * @author: alexandercd
 * @date: 22/septiembre/2021
 * @description: 
**/
import React from 'react';  

const GifGridItem = ({ title, url}) => {
    return (
        <div className="card animate__animated animate__flipInX"> 
            <div className="image">
                <img 
                    alt={title}
                    title={title}
                    src={url}
                />
            </div>
            <p>{title}</p>
        </div>
    )
} 

export default GifGridItem;