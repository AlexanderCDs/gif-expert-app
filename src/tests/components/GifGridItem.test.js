/**
 * @author: alexandercd
 * @date: 24/septiembre/2021
 * @description: 
**/
import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import GifGridItem from '../../components/GifGridItem';

describe('Test <GifGridItem />', () => { 
    const title = 'Un t√≠tulo';
    const img = 'https://localhost/mi-imagen.jpg';
    let wrapper = shallow( <GifGridItem />);

    beforeEach( () => {
        wrapper = shallow( <GifGridItem title={title} url={img} />);
    });
    
    test('should show <GifGridItem /> success', () => {  
        expect( wrapper ).toMatchSnapshot();
    });

    test('should have paragraph with title', () => {
        const paragraph = wrapper.find('p');
        expect( paragraph.text().trim() ).toBe( title );
    });

    test('should have image with alt and src', () => {
        const image = wrapper.find('img'); 
        expect( image.prop('src') ).toBe( img );
        expect( image.prop('alt') ).toBe( title );
    });

    test('should have animate__flipInX', () => {
        const div = wrapper.find('div').at(0);  
        const className = div.prop('className'); 
        expect( className.includes('animate__flipInX') ).toBe( true );
    });
     
});
