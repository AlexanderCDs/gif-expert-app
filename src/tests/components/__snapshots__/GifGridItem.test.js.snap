// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <GifGridItem /> should show <GifGridItem /> success 1`] = `
<div
  className="card animate__animated animate__flipInX"
>
  <div
    className="image"
  >
    <img
      alt="Un título"
      src="https://localhost/mi-imagen.jpg"
      title="Un título"
    />
  </div>
  <p>
    Un título
  </p>
</div>
`;
